indico/htdocs/js/indico/jquery/permissions_widget.js

```diff
@@ -0,0 +1,79 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2017 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    'use strict';
+
+    var permissionClasses = {
+        access: 'accept',
+        timetable: 'highlight',
+        edit: 'danger',
+        submit: 'warning'
+    };
+
+    $.widget('indico.permissionswidget', {
+        options: {
+            a: null,
+            b: {},
+            c: []
+        },
+        _update: function() {
+            this.$dataField.val(JSON.stringify(this.data));
+        },
+        _renderLabel: function(principal) {
+            var $labelBox = $('<div class="label-box">');
+            if (principal.type === 'role') {
+                var $code = $('<span class="role-code">').text(principal.code)
+                    .css({'border-color': '#' + principal.color, 'color': '#' + principal.color});
+                var $text = $('<span class="text-normal">').text(principal.name);
+                return $labelBox.append($('<span class="flexrow f-a-center">').append($code).append($text));
+            } else {
+                var iconClass = principal.type === 'user' ? 'icon-user' : 'icon-users';
+                var text = principal.type === 'user' ? principal.name : principal.id;
+                return $labelBox.append($('<span class="label-icon text-normal">').addClass(iconClass).text(text));
+            }
+        },
+        _renderPermissions: function(permissions) {
+            var $permissions = $('<div class="permissions-box flexrow f-a-center f-self-stretch">');
+            var $permissionsUl = $('<ul>').appendTo($permissions);
```

Comment:
I think `$permissionsList` is a better name - you'd also call something "fooLink" and not "fooA"